### PR TITLE
Fix coverage for Mutant::Meta namespace

### PR DESCRIPTION
**Pull request by mbj**
* Remove unparser invariant checking, does not yield enough value
  anymore as unparser is more stable these days

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb Assuming all green, this is ready for review.

File: lib/mutant/meta.rb [hunk 1]
```diff
@@ -15,15 +16,20 @@ class Example
       # @return [undefined]
       def self.add(&block)
         file = caller.first.split(':in', 2).first
-        ALL << DSL.run(file, block)
+        ALL << DSL.call(file, block)
       end
 
-      Pathname.glob(Pathname.new(__FILE__).parent.parent.parent.join('meta', '**/*.rb'))
+      Pathname.glob(Pathname.new(__dir__).parent.parent.join('meta', '*.rb'))
```
Comment (backus) [inline lib/mutant/meta.rb]: If you're adjusting this line could you use the shorthand `Pathname(__dir__)`?

File: lib/mutant/meta/example/dsl.rb [hunk 2]
```diff
@@ -9,18 +8,20 @@ class DSL
         # Run DSL on block
         #
         # @return [Example]
-        def self.run(file, block)
+        def self.call(file, block)
```
Comment (backus) [inline lib/mutant/meta/example/dsl.rb]: I'm curious, why did you rename this to `call`?

File: lib/mutant/meta/example/verification.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    class Example
+      # Example verification
+      class Verification
+        include Adamantium::Flat, Concord.new(:example, :mutations)
+
+        # Test if mutation was verified successfully
+        #
+        # @return [Boolean]
+        def success?
+          missing.empty? && unexpected.empty? && no_diffs.empty?
```
Comment (backus) [inline lib/mutant/meta/example/verification.rb]: I noticed you seem to have completely dropped the `unparser` method and `unparser.empty?`. Was that performing a check that is no longer necessary?

File: spec/unit/mutant/meta/example/verification_spec.rb [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+          no_diff: []
+        REPORT
+      end
+    end
+
+    context 'when there is extra generated node' do
+      let(:generated_nodes) { [s(:false), s(:nil)] }
+
+      specify do
+        should eql(strip_indent(<<-'REPORT'))
```
Comment (backus) [inline spec/unit/mutant/meta/example/verification_spec.rb]: Do you ever run your specs with a ruby version lower than 2.3? If not then I imagine this could use `<<~`

File: lib/mutant/meta.rb (hunk 1 above)
Comment (mbj) [inline lib/mutant/meta.rb]: I do not like using methods from Kernel, that infect every object without need.

File: lib/mutant/meta/example/dsl.rb (hunk 2 above)
Comment (mbj) [inline lib/mutant/meta/example/dsl.rb]: `.call` is the better convention. As Rubies own blocks / procs / lambdas follow these.

File: lib/mutant/meta/example/verification.rb (hunk 3 above)
Comment (mbj) [inline lib/mutant/meta/example/verification.rb]: Yes, this is outlined in the commit message.

File: spec/unit/mutant/meta/example/verification_spec.rb (hunk 4 above)
Comment (mbj) [inline spec/unit/mutant/meta/example/verification_spec.rb]: Yes 2.2 is still supported for development. I intend specs should pass under all versions mutant supports, as mutant supports 2.2 I cannot use 2.3 specific syntax.

File: lib/mutant/meta/example/verification.rb [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        end
+
+        # Error report
+        #
+        # @return [String]
+        def error_report
+          fail 'no error report on successful validation' if success?
+
+          YAML.dump(
+            'file' =>            example.file,
```
Comment (dkubb) [inline lib/mutant/meta/example/verification.rb]: I would suggest aligning the `=>` operators.

File: lib/mutant/meta/example/verification.rb [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        # @return [Boolean]
+        def success?
+          missing.empty? && unexpected.empty? && no_diffs.empty?
+        end
+
+        # Error report
+        #
+        # @return [String]
+        def error_report
+          fail 'no error report on successful validation' if success?
```
Comment (dkubb) [inline lib/mutant/meta/example/verification.rb]: You're ok with this failing with a `StandardError` exception? Not something more specific?

File: lib/mutant/meta/example/verification.rb [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+        # Mutation report
+        #
+        # @param [Array<Parser::AST::Node>] nodes
+        #
+        # @return [Array<Hash>]
+        def format_mutations(nodes)
+          nodes.map do |node|
+            {
+              'node' =>   node.inspect,
```
Comment (dkubb) [inline lib/mutant/meta/example/verification.rb]: Instance of https://github.com/mbj/mutant/pull/543/files#r55152676

File: lib/mutant/meta/example/verification.rb [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+          end
+        end
+
+        # No diff mutation report
+        #
+        # @return [Array, nil]
+        def no_diff_report
+          no_diffs.map do |mutation|
+            {
+              'node' =>   mutation.node.inspect,
```
Comment (dkubb) [inline lib/mutant/meta/example/verification.rb]: Instance of https://github.com/mbj/mutant/pull/543/files#r55152676

File: lib/mutant/meta/example/dsl.rb [hunk 9]
```diff
@@ -9,18 +8,20 @@ class DSL
...
           instance = new(file)
           instance.instance_eval(&block)
```
Comment (dkubb) [inline lib/mutant/meta/example/dsl.rb]: I think `#instance_eval` returns `self`. You could probably write this as:

```ruby
def self.call(file, block)
  new(file).instance_eval(&block).example
end
```

File: lib/mutant/meta/example/dsl.rb (hunk 2 above)
Comment (dkubb) [inline lib/mutant/meta/example/dsl.rb]: Why not use `&block` here to make it explicit that it's a Proc object.

**Comment by dkubb** [comment]
@mbj I did a quick first pass over this PR.

File: lib/mutant/meta/example/verification.rb (hunk 6 above)
Comment (mbj) [inline lib/mutant/meta/example/verification.rb]: Its a `RuntimeError`, and I do not want anyone to consider rescuing these errors, so I do not want to give any specific hint to do so via a dedicated exception class. This is only used from specs, and they *should* bubble to a failed example.

File: lib/mutant/meta/example/dsl.rb (hunk 9 above)
Comment (mbj) [inline lib/mutant/meta/example/dsl.rb]: It does not, it returns the last expression evaluated in block.

File: lib/mutant/meta/example/dsl.rb (hunk 2 above)
Comment (mbj) [inline lib/mutant/meta/example/dsl.rb]: Because the call side has to capture the block into an object already, as it needs to be evaluated somewhere else. It does not make sense for me to pass it as a block to capture it to pass it as a block again.

Also a real argument counts against the arity checks, where a block does not.

**Comment by mbj** [comment]
@dkubb Ready for the next pass.

File: lib/mutant/meta/example/dsl.rb (hunk 9 above)
Comment (dkubb) [inline lib/mutant/meta/example/dsl.rb]: I think the docs for `BasicObject#instance_eval` are incorrect then: http://ruby-doc.org/core-2.2.0/BasicObject.html#method-i-instance_eval
